extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -469,6 +473,19 @@ class MongoQueueProcessor {
             metricsExtension, () => {});
     }
 
+    /**
+     * For cases where we experience an error or skip an entry, we need to
+     * normalize pending metric. This means we will see pending metrics stuck
+     * above 0 and will need to bring those metrics down
+     * @param {string} location - location constraint name
+     * @return {undefined}
+     */
+    _normalizePendingMetric(location) {
+        const metric = { [location]: { ops: 1 } };
```

Comment:
🤔 I wonder if it would be more intuitive and extensible to use a generic `metricsTypePending` which allows passing of a negative value for decrementing. That way we shouldn't need a special type just for decrementing. Maybe there is a factor that is preventing this though that I'm not aware of.

Reply:
I changed it to have a type called "pendingOnly" and the caller can specify +/- values for ops and bytes. Since other metrics will report ops/bytes as positive values, its required to add negative values when reporting to the helper method. But other than that, it's a more flexible call. If normalizing requires a  positive ops value and a negative bytes value (if caller is batching), then thats possible with the change :+1: